src/lib/Scenes/Collection/Components/FeaturedArtists.tsx

```diff
@@ -69,17 +75,78 @@ export class FeaturedArtists extends React.Component<FeaturedArtistsProps, Featu
     return allArtists
   }
 
+  failedFollowChange() {
+    // callback for analytics purposes
+    this.setState({
+      isFollowedChanging: false,
+    })
+  }
+
+  successfulFollowChange() {
+    // callback for analytics purposes
+    Events.userHadMeaningfulInteraction()
+    this.setState({
+      isFollowedChanging: false,
+    })
+  }
+
+  handleFollowArtist(artist) {
+    const { slug, id, isFollowed } = artist
+    const { relay } = this.props
+    const { isFollowedChanging } = this.state
+
+    if (isFollowedChanging) {
+      return
+    }
+
+    this.setState(
+      {
+        isFollowedChanging: true,
+      },
+      () => {
+        commitMutation<FeaturedArtists_collectionMutation>(relay.environment, {
+          onCompleted: () => this.successfulFollowChange(),
+          mutation: graphql`
+            mutation FeaturedArtists_collectionMutation($input: FollowArtistInput!) {
+              followArtist(input: $input) {
+                artist {
+                  id
+                  isFollowed
+                }
+              }
+            }
+          `,
+          variables: {
+            input: {
+              artistID: slug,
+              unfollow: isFollowed,
+            },
+          },
+          optimisticResponse: {
```

Comment:
Ah, neat. I was going to ask why we don't see a spinner when we toggle the follow buttons in the entity header — I take it this is why?